File: Zipping and Unzipping files in .NET application/[VB]-Zipping and Unzipping files in .NET application/VB/NewNamePrompt.vb
Namespace ArchiveManager
	Partial Public Class NewNamePrompt
		Inherits Form
		Public Sub New(ByVal oldName As String)
			InitializeComponent()

			txtNewName.Text = oldName
		End Sub

		''' <summary>
		''' Gets the desired new name.
		''' </summary>
		Public Property NewName() As String
			Get
				Return txtNewName.Text
			End Get
			Set(ByVal value As String)
				txtNewName.Text = value
			End Set
		End Property

		''' <summary>
		''' Gets the desired new name caption.
		''' </summary>
		Public Property NewNameCaption() As String
			Get
				Return lbl.Text
			End Get
			Set(ByVal value As String)
				lbl.Text = value
			End Set
		End Property

		Private _rename As Boolean = True
		''' <summary>
		''' Gets the desired new name caption.
		''' </summary>
		Public Property Rename() As Boolean
			Get
				Return _rename
			End Get
			Set(ByVal value As Boolean)
				_rename = value
			End Set
		End Property

		''' <summary>
		''' Handles the OK button's Click event.
		''' </summary>
		''' <param name="sender">The button object.</param>
		''' <param name="e">The event arguments.</param>
		Private Sub btnOk_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnOk.Click
			If _rename Then
				' Check for invalid characters.
				If NewName.Contains("/") OrElse NewName.Contains("\") Then
					MessageBox.Show(Messages.InvalidCharacters, Me.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop)
					Return
				End If
			End If

			Me.DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub
	End Class
End Namespace

File: Zipping and Unzipping files in .NET application/[C#]-Zipping and Unzipping files in .NET application/C#/NewNamePrompt.cs
using System;
using System.Windows.Forms;

namespace ArchiveManager
{
    public partial class NewNamePrompt : Form
    {
        public NewNamePrompt(string oldName)
        {
            InitializeComponent();

            txtNewName.Text = oldName;
        }

        /// <summary>
        /// Gets the desired new name.
        /// </summary>
        public string NewName
        {
            get
            {
                return txtNewName.Text;
            }
            set
            {
                txtNewName.Text = value;
            }
        }

        /// <summary>
        /// Gets the desired new name caption.
        /// </summary>
        public string NewNameCaption
        {
            get
            {
                return lbl.Text;
            }
            set
            {
                lbl.Text = value;
            }
        }

        bool _rename = true;
        /// <summary>
        /// Gets the desired new name caption.
        /// </summary>
        public bool Rename
        {
            get
            {
                return _rename;
            }
            set
            {
                _rename = value;
            }
        }

        /// <summary>
        /// Handles the OK button's Click event.
        /// </summary>
        /// <param name="sender">The button object.</param>
        /// <param name="e">The event arguments.</param>
        private void btnOk_Click(object sender, EventArgs e)
        {
            if (_rename)
            {
                // Check for invalid characters.
                if (NewName.Contains("/") || NewName.Contains("\\"))
                {
                    MessageBox.Show(Messages.InvalidCharacters, this.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop);
                    return;
                }
            }

            this.DialogResult = DialogResult.OK;
        }
    }
}